bapsflib/_hdf/maps/controls/tests/fauxnixyz.py

```diff
@@ -0,0 +1,311 @@
+        # create numpy array
+        dset_name = 'Run time list'
+        shape = (self.knobs.sn_size, )
+        dtype = np.dtype([
+            ('Shot number', np.int32),
+            ('Configuration name', np.bytes_, 120),
+            ('motion_index', np.int32),
+            ('x', np.float64),
+            ('y', np.float64),
+            ('z', np.float64),
+            ('r', np.float64),
+            ('phi', np.float64),
+            ('theta', np.float64),
+        ])
+        data = np.ndarray(shape=shape, dtype=dtype)
+
+        # assign shot numbers
+        data['Shot number'] = \
+            np.arange(1, shape[0] + 1, 1,
+                      dtype=data.dtype['Shot number'].type)
+
+        # assign motion lists (aka 'Configuration name')
+        # assign motion_index
+        start = 0
+        for ml in self._ml:
+            # determine stop index
+            stop = start + ml['size']
+
+            # insert values
+            data['Configuration name'][start:stop:1] = ml['name']
+            data['motion_index'][start:stop:1] = \
+                np.arange(0, ml['size'], 1,
+                          dtype=data.dtype['motion_index'].type)
+
+            # move start index
+            start = stop
+
+        # fill position fields
+        # TODO: create a real fill, opposed to zeros
+        #
+        data['x'].fill(0.0)
+        data['y'].fill(1.0)
+        data['z'].fill(2.0)
+        data['r'].fill(3.0)
+        data['phi'].fill(np.pi/10.0)
+        data['theta'].fill(np.pi/20.0)
+
+        # create dataset
+        self.create_dataset(dset_name, data=data)
+
+        # add to configs
+        self.configs['config01']['dset name'] = dset_name
+
+    def _add_motionlist_groups(self):
+        """Add motion list groups"""
+        # determine possible data point arrangements for motion lists
+        # 1. find divisible numbers of sn_size
+        # 2. find (Nx, Ny, Nz) combos for each dataset
+        sn_size_for_ml = []
+        NN = []
+        if self.knobs.n_motionlists == 1:
+            # set shot number size per for each motion list
+            sn_size_for_ml.append(self.knobs.sn_size)
+        else:
+            # set shot number size per for each motion list
+            sn_per_ml = int(math.floor(
+                self.knobs.sn_size / self.knobs.n_motionlists))
+            sn_remainder = (self.knobs.sn_size
+                            - ((self.knobs.n_motionlists - 1)
+                               * sn_per_ml))
+            sn_size_for_ml.extend([sn_per_ml]
+                                  * (self.knobs.n_motionlists - 1))
+            sn_size_for_ml.append(sn_remainder)
+
+        # build NN of each motion list
+        for i in range(self.knobs.n_motionlists):
+            # find divisible numbers
+            sn_div = []
+            for j in range(sn_size_for_ml[i]):
+                if sn_size_for_ml[i] % (j + 1) == 0:
+                    sn_div.append(j + 1)
+
+            # build (Nx, Ny, Nz)
```

Comment:
the `build (Nx, Ny, Nz)` code block (replaces lines 220-226) does not ensure `Nx * Ny * Nz = number of shots`

```suggestion
            # -- build NN = [(Nx, Ny, Nz), ..,] --
            # define Nx
            sn_div_index = random.randint(0, len(sn_div) - 1)
            Nx = sn_div[sn_div_index]

            # define Ny
            sn_div.clear()
            sn_left = int(sn_size_for_ml[i] / Nx)
            for j in range(sn_left):
                if sn_left % (j + 1) == 0:
                    sn_div.append(j + 1)
            sn_div_index = random.randint(0, len(sn_div) - 1)
            Ny = sn_div[sn_div_index]

            # define Nz
            Nz = int(sn_left / Ny)
            NN.append((Nx, Ny, Nz))
```